src/Microsoft.Sbom.Api/Executors/Spdx2SerializationStrategy.cs

```diff
@@ -0,0 +1,162 @@
+    {
+        if (config.MetadataBuilder.TryGetPackageArrayHeaderName(out var headerName))
+        {
+            config.JsonSerializer.StartJsonArray(headerName);
+            elementsSupportingConfigs.Add(config);
+        }
+    }
+
+    public bool AddToRelationshipsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        if (config.MetadataBuilder.TryGetRelationshipsHeaderName(out var headerName))
+        {
+            config.JsonSerializer.StartJsonArray(headerName);
+            return true;
+        }
+
+        return false;
+        }
+
+    public void AddToExternalDocRefsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        if (config.MetadataBuilder.TryGetExternalRefArrayHeaderName(out var headerName))
+        {
+            config.JsonSerializer.StartJsonArray(headerName);
+            elementsSupportingConfigs.Add(config);
+        }
+    }
+
+    public void AddHeadersToSbom(ISbomConfigProvider sbomConfigs)
+    {
+        sbomConfigs.ApplyToEachConfig(config =>
+            config.JsonSerializer.WriteJsonString(
+                config.MetadataBuilder.GetHeaderJsonString(sbomConfigs)));
+    }
+
+    public async Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
+        ISbomConfig sbomConfig,
+        string spdxManifestVersion,
+        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
+        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
+        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
+        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    {
+        fileArrayGenerator.SbomConfig = sbomConfig;
+        packageArrayGenerator.SbomConfig = sbomConfig;
+        relationshipsArrayGenerator.SbomConfig = sbomConfig;
+        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
+
+        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
+
+        var errors = new List<FileValidationResult>();
+
+        await WriteFiles(fileArrayGenerator);
+
+        await WritePackages(packageArrayGenerator);
+
+        await WriteExternalDocRefs(externalDocumentReferenceGenerator);
+
+        await WriteRelationships(relationshipsArrayGenerator);
+
+        return errors;
+    }
+
+    /// <summary>
+    /// Write to Files section
+    /// </summary>
+    /// <param name="fileArrayGenerator"></param>
+    /// <returns></returns>
+    private async Task WriteFiles(IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator)
+    {
+        var filesGenerateResult = await fileArrayGenerator.GenerateAsync();
```

Comment:
Variable name didn't change when `GenerateResult` type was renamed to `GenerationResult`